pilot/pkg/networking/core/v1alpha3/listener_waypoint.go

```diff
@@ -199,7 +176,10 @@ func (lb *ListenerBuilder) buildWaypointInboundTerminateConnect(svcs map[host.Na
 	h.Http2ProtocolOptions = &core.Http2ProtocolOptions{
 		AllowConnect: true,
 	}
-	h.HttpFilters = append([]*hcm.HttpFilter{xdsfilters.ConnectBaggageFilter}, h.HttpFilters...)
+	h.HttpFilters = append([]*hcm.HttpFilter{
+		xdsfilters.ConnectBaggageFilter,
+		xdsfilters.ConnectAuthorityFilter,
```

Comment:
Could you add some comment? Is this filter where authz can be handled?

Reply:
No, these to propagate :authority and baggage tunnel headers to the inner stream.

Reply:
Could you explain how L7 authz will be enforced with this change?

Reply:
This is not the PR for L7 authz. That's to be discussed separately in https://github.com/istio/istio/issues/43009, and the tests are disabled until we have a design.

This PR is needed so we have a solid base to scale Envoy. Without it, there's no path to scalable implementation of authz.

Reply:
Thanks, makes sense, so this change is based on last Wed's upstream meeting where we won't allow users to select authz based on workload label and it needs to be bound to the waypoint(s).

I assume we'll continue to use Envoy http filter (rbac filter) to do authz enforcement.

Reply:
Yes, it's based on the decision not to use workload labels. Using Envoy RBAC makes sense. The current API proposal requires adding a new field I think, so we can't modify tests easily until the API change is merged.